### PR TITLE
datastore: track bucket count in store stats & mem panel

### PR DESCRIPTION
**Pull request by teh-cmc**
image

### CODE REVIEW

File: crates/re_viewer/src/ui/memory_panel.rs [hunk 1]
```diff
@@ -175,7 +175,45 @@ impl MemoryPanel {
...
+                let DataStoreStats { config, .. } = store_stats;
+
+                ui.label(egui::RichText::new("Limits").italics());
+                ui.label("Row limit");
+                ui.label("Size limit");
+                ui.end_row();
+
+                let label_rows = |ui: &mut egui::Ui, num_rows| {
+                    if num_rows == u64::MAX {
+                        ui.label("+∞")
```
Comment (Wumpf) [inline crates/re_viewer/src/ui/memory_panel.rs:190]: ... well technically not inf :D

**Review by Wumpf — APPROVED** [review]
niiice!!

I'd be inclined to call `temporal` rather `regular` tho 🤷

**Comment by Wumpf** [comment]
Actually, what *is* taking all the memory in that application if it isn't the store?

**Comment by teh-cmc** [comment]
> I'd be inclined to call `temporal` rather `regular` tho 🤷 

This terminology is used everywhere else, so I'd rather stay consistent for now.

> Actually, what is taking all the memory in that application if it isn't the store?

I haven't looked in detail, but this program ingests 600k index rows worth of data, so I imagine all of the datastructures in the viewer that scale linearly with the number of messages?

**Comment by emilk** [comment]
> Actually, what _is_ taking all the memory in that application if it isn't the store?

In the `clock` example, is the `LogMsg`:es stored in the viewer. Since we don't batch the log calls, we produce one huge `ArrowMsg` per log call.

You can see this with `RERUN_TRACK_ALLOCATIONS=1 cargo r -p rerun -- ../clock_50k.rrd`

**Pull request merged**